tools/dev_qa.py

```diff
@@ -24,67 +24,64 @@
     "QA_ROOT", os.path.join(os.path.expanduser("~"), "buildbuddy-qa")
 )
 
+API_KEY = os.environ.get(
+    "BB_API_KEY",
+)
+
 REPO_CONFIGS = [
     {
         "name": "buildbuddy",
         "repo_url": "https://github.com/buildbuddy-io/buildbuddy",
         "command": """
             bazel test //... \
                 --config=remote-dev \
-                --flaky_test_attempts=3 \
-                --build_metadata=TAGS=dev-qa
-        """,
-    },
-    {
-        "name": "bazel-gazelle",
-        "repo_url": "https://github.com/bazelbuild/bazel-gazelle",
-        "command": """
-            bazel build //... \
                 --remote_executor=remote.buildbuddy.dev \
                 --remote_cache=remote.buildbuddy.dev \
                 --bes_backend=remote.buildbuddy.dev \
                 --bes_results_url=https://app.buildbuddy.dev/invocation/ \
-                --remote_timeout=10m \
-                --build_metadata=TAGS=dev-qa
-        """,
+                --flaky_test_attempts=3 \
+                --build_metadata=TAGS=dev-qa \
+                --remote_header=x-buildbuddy-api-key={}
+        """.format(API_KEY),
     },
     {
-        "name": "abseil-cpp",
-        "repo_url": "https://github.com/abseil/abseil-cpp",
+        "name": "bazel-gazelle",
+        "repo_url": "https://github.com/bazelbuild/bazel-gazelle",
         "command": """
             bazel build //... \
-                --cxxopt="-std=c++14" \
                 --remote_executor=remote.buildbuddy.dev \
                 --remote_cache=remote.buildbuddy.dev \
                 --bes_backend=remote.buildbuddy.dev \
                 --bes_results_url=https://app.buildbuddy.dev/invocation/ \
                 --remote_timeout=10m \
+                --build_metadata=TAGS=dev-qa \
+                --noenable_bzlmod \
                 --extra_execution_platforms=@buildbuddy_toolchain//:platform \
                 --host_platform=@buildbuddy_toolchain//:platform \
                 --platforms=@buildbuddy_toolchain//:platform \
                 --crosstool_top=@buildbuddy_toolchain//:toolchain \
-                --build_metadata=TAGS=dev-qa
-        """,
+                --remote_header=x-buildbuddy-api-key={}
+        """.format(API_KEY),
     },
     {
-        "name": "bazel",
```

Comment:
nit: instead of removing `bazel` we could also pin at a known good commit and bazel version. I think in general we probably want to pin all of these to a known good commit (plus a bazel version if a `.bazelversion` is missing from the repo), so that if these repos break then we don't have to block the release

Reply:
Done - pinned all the commit shas to current HEAD. For the bazel repo, pinned to the parent commit before they cleared their WORKSPACE file. Also pinned bazel version to 7.0.2

Was going back and forth on whether we should pin the bazel version, because we might want to know if builds are breaking on a later bazel version. But I think pinning it will make things more stable, and we can find that out other ways

Reply:
I'm actually having trouble getting `bazel` repo to build on an older commit. I want to make sure these fixes get in before the next release, so just going to remove bazel for now. We'll still have a go, c++, and python test build